portfolio/src/main/webapp/script.js

```diff
@@ -77,6 +77,32 @@ function openTab(evt, sectionName) {
 	evt.currentTarget.className += " active";
 }
 
+/**
+ * Gets the authentication status of user and other relevant user info 
+ */
+
+async function preparePage() {
+    const response = await fetch('/auth');
+    const currentUser = await response.json();
+    if (currentUser.loggedIn && currentUser.nickname == "") {
+        currentUser.nickname = prompt("Please enter your desired display name: ");
+        const params = new URLSearchParams();
+        params.append('nickname', currentUser.nickname);
+        await fetch('/auth', {method: 'POST', body: params});
```

Comment:
You might want to check the response to make sure it's a 200.  Otherwise I think it will appear to work, but nothing will actually happen.